nnvm/src/compiler/precompute_prune.cc

```diff
@@ -75,17 +75,17 @@ nnvm::Graph PrecomputePrune(nnvm::Graph src) {
     }
   });
 
-  // nothing being pruned.
-  if (non_var_edge == 0) {
-    return src;
-  }
-
   for (auto& e : src.outputs) {
     if (pruned.count(e.node.get())) {
       e = replace_pruned_entry(e);
     }
   }
 
+  // nothing being pruned.
+  if (non_var_edge == 0) {
```

Comment:
In some cases, this return is still too early. For example, `replace_pruned_entry` may still be called to replace a parameter node and modify its name by appending a version number. This change of the parameter name must be propagated to the frontend by setting the attribute "output_names" to the graph. Therefore, simply removing this block would solve the issue.